mungegithub/mungers/approval-handler.go

```diff
@@ -90,8 +90,31 @@ func (h *ApprovalHandler) Munge(obj *github.MungeObject) {
 	}
 
 	approverSet := sets.String{}
+	if obj.Issue.User.Name != nil {
+		approverSet.Insert(*obj.Issue.User.Name)
+	}
 
 	// from oldest to latest
+	approverSet = approverSet.Union(createApproverSet(comments))
+	//Checks that no valid approver has commented to cancel the approve label since last commit
+	//Checks that all files have an approver in the approverSet
+	needsApproval := h.getApprovalNeededFiles(files, approverSet)
+
+	//every file has been approved by a valid approver
+	if needsApproval.Len() > 0 {
+		//need to selectively write a comment explaining why approved not added
+		glog.Infof("Canceling the approve label because these files %v have no valid approvers", needsApproval)
+		if obj.HasLabel(approvedLabel) {
+			obj.RemoveLabel(approvedLabel)
+		}
+	} else {
+		obj.AddLabel(approvedLabel)
```

Comment:
Above you test if the label is already there, here you don't check if it's already there or not. You probably should